generator/src/test/java/com/scottlogic/deg/generator/cucumber/OfLength.feature

```diff
@@ -1,593 +1,455 @@
-
-
-  Scenario Outline: Running an 'ofLength' request that includes a negation of a valid numeric length should be successful
-    Given foo is anything but of length <length>
-    And foo is in set:
-      | "a"  |
-      | "aa" |
-    Then the following data should be generated:
-      | foo        |
-      | null       |
-      | <expected> |
-    And the following data should not be included in what is generated:
-      | foo           |
-      | <notExpected> |
-  Examples:
-      | length | expected    | notExpected |
-      | 1      | "aa"        | "a"         |
-      | 2      | "a"         | "aa"        |
-
-
-  Scenario Outline: Running an 'ofLength' request on a roman alphabet character string value and white space should be successful
-    Given foo is of length <length>
-    And foo is in set:
-      | "a"          |
-      | "  10  10  " |
-    Then the following data should be generated:
-      | foo        |
-      | null       |
-      | <expected> |
-    Examples:
-      | length | expected     |
-      | 10     | "  10  10  " |
-
-
-  Scenario: Running an 'ofLength' request that includes a numeric value = 0 on an empty string should be successful
-    Given foo is of length 0
-    And foo is in set:
-      | ""  |
-      | "a" |
-    Then the following data should be generated:
-      | foo  |
-      | null |
-      | ""   |
+Background:
+     Given the generation strategy is full
+       And there is a field foo
+       And foo is of type "string"
+
+Scenario Outline: Running an 'ofLength' request on a roman alphabet character string should be successful
+     Given foo is of length <length>
+       And foo is in set:
+         | "a"  |
+         | "aa" |
+     Then the following data should be generated:
+       | foo        |
+       | null       |
+       | <expected> |
+       And the following data should not be included in what is generated:
+         | foo           |
+         | <notExpected> |
+     Examples:
+       | length | expected | notExpected |
+       | 1      | "a"      | "aa"        |
+       | 2      | "aa"     | "a"         |
+       | 1.0    | "a"      | "aa"        |
```

Comment:
Have we got a consistent tabbing style for cucumber tests? The indentation looks a bit off in this test (this applies to multiple scenarios in the file but I won't repeat the same comment everywhere.

Reply:
There is a standard format and this has been applied
- Scenario: no indent
- Given / When / Then / Examples: indented by 5
- And: indented by 2 from the Given / When / Then
- Tables: indented by 2 from the Given / When / Then / Examples

Reply:
Removed not expected sections of these tests to provide better formatted view

Reply:
Ok dokes, this is the first time I've seen this format in a pull request - I'm not sure it's been consistently applied across everything else. It seems a little complex to me but if it's agreed by the team then that's fine